src/LIBSVM.jl

```diff
@@ -329,14 +370,15 @@ function svmtrain(
     ninstances = size(X, 2)
 
     # Construct SVMProblem
-    if kernel == Kernel.Precomputed
-        (nodes, nodeptrs) = gram2nodes(X)
-
-        # This is necessary to construct SupportVectors correctly
-        # The struct needs to contain the indices of support vectors
-        X = (1:size(X, 1))'
+    if !isa(kernel, Kernel.KERNEL)
+        gram = data2gram(kernel, X)
+        (nodes, nodeptrs) = gram2nodes(gram)
     else
-        (nodes, nodeptrs) = instances2nodes(X)
+        if kernel == Kernel.Precomputed
+            (nodes, nodeptrs) = gram2nodes(X)
+        else
+            (nodes, nodeptrs) = instances2nodes(X)
+        end
     end
```

Comment:
```suggestion
    nodes, nodeptrs = if !(kernel isa Kernel.KERNEL)
        gram2nodes(data2gram(kernel, X))
    elseif kernel == Kernel.Precomputed
        gram2nodes(X)
    else
        instances2nodes(X)
    end
```

Reply:
Would 
```julia
    if !(kernel isa Kernel.KERNEL)
        nodes, nodeptrs = gram2nodes(data2gram(kernel, X))
    elseif kernel == Kernel.Precomputed
        nodes, nodeptrs = gram2nodes(X)
    else
       nodes, nodeptrs =  instances2nodes(X)
    end
```
be acceptable to you? I think it's a bit easier to read & doesn't take any more lines but otherwise I don't feel strongly about it.

Sidenote: I didn't know about the `kernel isa Kernel.KERNEL` syntax, that seems more elegant than `isa(kernel, Kernel.KERNEL)`. In any case, I would also try to be consistent within the PR but I'd be fine with either style, really.

Reply:
ha, :ok_hand: